app/grandchallenge/challenges/models.py

```diff
@@ -1424,10 +1424,13 @@ class TaskResponsiblePartyChoices(models.TextChoices):
 
 
 class OnboardingTaskQuerySet(models.QuerySet):
-    def with_overdue_status(self):
+    def with_overdue_status(self, soon_delta=None):
```

Comment:
Why don't we hardcode `soon_delta` to be 7 days? That way the emails are consistent with the UI - as is people will get an email saying tasks are due soon but they may not be highlighted in the table. I don't think this is something that needs to be changable and is anyway tightly coupled to the crontab of the task.

Reply:
I get why you'd want to make the emails consistant with the UI. Making the `soon_delta` 7 days would help with it somewhat but there is the periodicity of the crontab and the on-request of the UI display which will always cause some inconsistenties. With 7 days you might see warning in the UI show up before the email was sent. (Which still holds for the current 3 days but less so). For consistency we'd have to make the email and the UI change to coincide, which is a bit complex(er).

Having said that, if we make it 7 days we will ensure we only sent an email when the UI also shows the soon warnings. That does push things to be more consistent. I'll update it.